### DISCUSSION TITLE
Copilot does not know about .net7 or .net8

### DISCUSSION THREAD

**Discussion by Hefaistos68** [community/community#124847] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I just got the following answer from Copilot when trying to get information on .net 8:
`I apologize for the confusion, but as an AI developed by OpenAI, I'm currently not able to provide real-time updates or information beyond my last training data in October 2021. As of that time, the latest stable version of .NET was .NET 6.0, which was released in November 2021.`

So I am paying 10 bucks a month for answers based on 3 years old data? Isn't it the least to expect that Copilot supports the latest versions of all .net components since its a "programming assistant" built in to Visual Studio?????

Edit: And why the heck does the web version of copilot (copilot.microsoft.com) know how to lookup info on the web and create an satisfactory answer from it?

**Comment by ayim** [comment]
@Hefaistos68 astute observation!
Yes, the web version of Copilot has implemented functionality to lookup info on the web.

Are you in Visual Studio?
We are actively investigating, I would encourage you to follow this issue:
https://developercommunity.visualstudio.com/t/Copilot-Chat-lastest-info-about-dotnet-i/10625798

![image](https://github.com/community/community/assets/10951869/ab32c7d5-c81c-4405-b839-55be4e244858)

-- 
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
github: ayim | aayim@microsoft.com

**Reply by Hefaistos68** [reply]
Believe me, i am following everything copilot related. Am using it since closed beta, and in general am very satisfied with it. Except for its lack of updated information.

**Comment by ayim** [comment]
Related: https://github.com/orgs/community/discussions/115874

**Comment by rs38** [comment]
what is your latest training data?

GitHub Copilot
I don't have access to my training data, but I was trained on a mixture of licensed data, data created by human trainers, and publicly available data. My training only includes information up until September 2021, so I may not have information on more recent events or developments. How can I assist you with your programming needs?

As of my last update, .NET 8 has not been released. 

this is not very helpful in end of 2024

**Comment by bluuf** [comment]
We're in 2025 and Copilot has no official knowledge of any supported dotnet version.... It has some basic knowledge about dotnet 8 (but since it's based on the RC versions and not the final version it's useless).
Dotnet Aspire is unknown to Github copilot, newer language features in C# are unknown: it really starts to feel like a scam asking us to pay for copilot since the data is severely outdated (and I'm not even talking about the many mistakes it makes in simple stuff which it should actually have knowlegde of, like setting up authentication with OIDC, 9/10 times the suggestions omit crucial steps you need to configure in your project)

**Comment by rs38** [comment]
just checked with Github Copilot Chat (Claude and Gtp4o) and it's aware of Aspire:
![image](https://github.com/user-attachments/assets/9d297b98-9839-4c97-83c6-cee83a9339d2)

**Reply by bluuf** [reply]
wow, didn't notice that they added the Claude options to the chat (which does know stuff about Aspire). Using gpt-4o I get "As of my training data up to October 2023, there is no specific framework or tool named "dotnet aspire" that is widely recognized in the .NET ecosystem."

Good that they've added Claude, this makes the tool a bit more usefull for me
